examples/events/README.md

```diff
@@ -292,4 +111,129 @@ function computeDropzoneProps(props, node) {
 <Parent />
 ```
 
-Note that the sort of behavior illustrated above can lead to some confusion. For example, the `onDragLeave` callback of the parent will not be called if the callback for the same event is invoked on the child. This happens because we invoked `stopPropagation()` on the event from the child.
+Note that internally we use `event.stopPropagation()` to achieve the behavior illustrated above, but this comes with its own [drawbacks](https://javascript.info/bubbling-and-capturing#stopping-bubbling).
+
+If you'd like to selectively turn off the default dropzone behavior for `onClick`, `onKeyDown` (both open the file dialog), `onFocus` and `onBlur` (sets the `isFocused` state) and drag events; use the `{noClick}`, `{noKeyboard}` and `{noDrag}` properties:
+```jsx harmony
+import React, {useCallback, useReducer} from 'react';
+import {useDropzone} from 'react-dropzone';
+
+const initialEvtsState = {
+  noClick: true,
+  noKeyboard: true,
+  noDrag: true,
+  files: []
+};
+
+function Events(props) {
+  const [state, dispatch] = useReducer(reducer, initialEvtsState);
+  const createToggleHandler = type => () => dispatch({type});
+
+  const onDrop = useCallback(files => dispatch({
+    type: 'setFiles',
+    payload: files
+  }), []);
```

Comment:
Since you're not listening to changes on any dependencies, isn't this overkill? Could this be rewritten as a simple function with type defined in the ts definition: 

`onDrop?(acceptedFiles: File[], rejectedFiles: File[], event: DropEvent): void`

```
const onDrop = files => dispatch({
    type: 'setFiles',
    payload: files
  });
```

My overarching goal here is simplicity and readability for future programmers. Even if they're adept at react hooks, why parse complexity if it could be simple.

Reply:
Using the `useCallback(..., [])` hook that way creates a memoized version of the fn, which means we avoid re-rendering (because internally `useDropzone()` also caches and has `onDrop` as dependency).

But I guess it's not that important and we could remove it.